cyclient/scan_client.py

```diff
@@ -45,6 +49,51 @@ def zipped_file_scan(self, scan_type: str, zip_file: InMemoryZip, scan_id: str,
         except Exception as e:
             self._handle_exception(e)
 
+    def zipped_file_scan_async(self, zip_file: InMemoryZip, scan_type: str, scan_parameters: dict,
+                               is_git_diff: bool = False) -> models.ScanInitializationResponse:
+        url_path = f"{self.SCAN_SERVICE_CONTROLLER_PATH}/{scan_type}/repository"
+        files = {'file': ('multiple_files_scan.zip', zip_file.read())}
+        try:
+            response = self.cycode_client.post(url_path=url_path,
+                                               data={'is_git_diff': is_git_diff,
+                                                     'scan_parameters': json.dumps(scan_parameters)},
+                                               files=files)
+            return models.ScanInitializationResponseSchema().load(response.json())
+        except Exception as e:
+            self._handle_exception(e)
+
+    def get_scan_details(self, scan_id: str) -> models.ScanDetailsResponse:
+        url_path = f"{self.SCAN_SERVICE_CONTROLLER_PATH}/{scan_id}"
+        try:
+            response = self.cycode_client.get(url_path=url_path)
+            return models.ScanDetailsResponseSchema().load(response.json())
+        except Exception as e:
+            self._handle_exception(e)
+
+    def get_scan_detections(self, scan_id: str) -> List[dict]:
+        detections = []
+        page_number = 0
+        page_size = 200
+        last_response_size = 0
+        try:
+            while page_number == 0 or last_response_size == page_size:
+                url_path = f"{self.DETECTIONS_SERVICE_CONTROLLER_PATH}?scan_id={scan_id}&page_size={page_size}&page_number={page_number}"
+                response = self.cycode_client.get(url_path=url_path)
```

Comment:
move this logic to another function `get_detections_paginated` or better name and this func should as arg of filters that its type is dict
```suggestion
                url_path = f"{self.DETECTIONS_SERVICE_CONTROLLER_PATH}?scan_id={scan_id}&page_size={page_size}&page_number={page_number}"
                response = self.cycode_client.get(url_path=url_path)
```

Reply:
regarding the name it was something like you recommended and Maor ask to move it get_scan_detections

Reply:
I dont think that we should extract the filter to argument at this point.
_get_scan_result should create filter and for now it's the only method that use it..